### PR TITLE
Fixes for compiler warnings

### PR DESCRIPTION
**Pull request by yvaind**
Fixed some warnings that would be triggered in downstream packages including header files from this repo when building with the '-Wall -Wextra -Wpedantic' compiler flags.

### CODE REVIEW

**Review by goodfaiter — APPROVED** [review]

**Comment by yvaind** [comment]
@goodfaiter I'm sorry, but you need to unapprove and review again. I added about twice as many new changes and we need to discuss a few things. Since these may be relevant for deciding what to do with other open-sourced packages, I also added you @remod as a reviewer.

- There are at least 3 different ways errors are logged/handled: std::cerr, std::runtime_error, and CHECK (a google macro defined in logging.h). I think we should adhere to one standard, the ideal one maybe being MELO. However, this would introduce another dependency.
- There are tons of virtual and overriden functions that are not implemented and will not even trigger a runtime error if used. There are a few places where an error is printed, but not in a consistent manner (see previous item). IMO all of them should print an error and we need to decide which way to go.
- The override specifier is not used a single time, although plenty of virtual functions get overriden.
- Adding clang tooling would be nice. But it also means an additional, new dependency.
- There are files that are not compiled when building curves, but only when building curves_ros. I fear that there may be even more files inside curves that only get compiled when used by another package.
- There are certainly more issues...

File: curves/CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,8 @@
 cmake_minimum_required(VERSION 2.8.3)
```
Comment (remod) [inline curves/CMakeLists.txt]: 3.1.3 for CMAKE_CXX_STANDARD

Comment (yvaind) [reply curves/CMakeLists.txt]: Done

**Review by remod — APPROVED** [review]

**Comment by yvaind** [comment]
I discussed with @remod and as you suggested @goodfaiter, we think it's best to address above issues (#81 , #80 , #79 , #78 ) in separate PRs.

**Review by goodfaiter — APPROVED** [review]

**Review by harmishhk — APPROVED** [review]

**Pull request merged**